packages/xstate-wallet/src/logger.ts

```diff
@@ -13,13 +13,24 @@ const name = 'xstate-wallet';
 const destination =
   LOG_TO_FILE && !IS_BROWSER_CONTEXT ? pino.destination(LOG_DESTINATION) : undefined;
 
-// If we are in a browser, but we want to LOG_TO_FILE, we assume that the
-// log statements are meant to be stored as JSON objects
-// So, we log serialized objects, appending the name (which the pino browser-api appears to remove?)
-const browser =
-  LOG_TO_FILE && IS_BROWSER_CONTEXT
-    ? {write: o => console.log(JSON.stringify({...o, name}))}
-    : undefined;
+const serializeLogEvent = o => JSON.stringify({...o, name});
+
+let browser: any = IS_BROWSER_CONTEXT
+  ? {
+      transmit: {
+        send: (_, logEvent) =>
+          window.parent.postMessage(
```

Comment:
Might be worthwhile adding a small comment explaining the use of `postMessage` as it can be easy to miss.